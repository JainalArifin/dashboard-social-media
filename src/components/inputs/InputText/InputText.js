import * as React from "react";
import TextField from "@mui/material/TextField";

export default function InputText({
  label,
  errorMessage,
  isError = false,
  width = 300,
  ...rest
}) {
  return (
    <>
      <TextField
        style={{ width: width }}
        helperText={errorMessage}
        id="input-text"
        label={label}
        color={isError && "error"}
        error={isError}
        {...rest}
      />
    </>
  );
}
