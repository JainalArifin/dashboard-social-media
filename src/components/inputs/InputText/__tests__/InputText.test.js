import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import InputText from "../InputText";
import { Formik, Field } from "formik";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Formik>
        <Field name="fname" component={InputText} isError={false} />
      </Formik>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
