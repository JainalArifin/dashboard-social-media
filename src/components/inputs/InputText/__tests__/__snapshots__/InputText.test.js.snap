// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  field={
    Object {
      "name": "fname",
      "onBlur": [Function],
      "onChange": [Function],
      "value": undefined,
    }
  }
  form={
    Object {
      "dirty": false,
      "errors": Object {},
      "getFieldHelpers": [Function],
      "getFieldMeta": [Function],
      "getFieldProps": [Function],
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialErrors": Object {},
      "initialStatus": undefined,
      "initialTouched": Object {},
      "initialValues": undefined,
      "isSubmitting": false,
      "isValid": true,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validateOnMount": false,
      "values": undefined,
    }
  }
  style={
    Object {
      "width": 300,
    }
  }
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="input-text"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
